### PR TITLE
Use `Record::get` instead of `Value` functions

### PR DESCRIPTION
**Pull request by IanManske**
# Description
Where appropriate, this PR replaces instances of `Value::get_data_by_key` and `Value::follow_cell_path` with `Record::get`. This avoids some unnecessary clones and simplifies the code in some places.

### CODE REVIEW

**Review by sholderbach — APPROVED** [review]
Looking good!

Gotta watch out for potential merge conflicts but we should be moving along with the record stuff for this release

File: crates/nu-table/src/types/expanded.rs [hunk 1]
```diff
@@ -116,10 +116,11 @@ fn expanded_table_list(input: &[Value], cfg: Cfg<'_>) -> TableResult {
...
             let index = row + cfg.opts.row_offset;
-            let text = matches!(item, Value::Record { .. })
-                .then(|| {
-                    lookup_index_value(item, cfg.opts.config).unwrap_or_else(|| index.to_string())
-                })
+            let text = item
+                .as_record()
+                .ok()
+                .and_then(|val| val.get(INDEX_COLUMN_NAME))
+                .map(|value| value.into_string("", cfg.opts.config))
```
[the remark this answers is not in the document]
Comment (IanManske) [reply crates/nu-table/src/types/expanded.rs:123]: Not sure, I was just matching the logic in the old `lookup_index_value` function removed below.

File: crates/nu-explore/src/nu_common/value.rs [hunk 2]
```diff
@@ -152,30 +150,15 @@ fn create_table_for_record(headers: &[String], items: &[Value]) -> Vec<Vec<Value
...
-        item => item.clone(),
+    if let Value::Record { val, .. } = item {
+        headers
+            .iter()
+            .map(|col| val.get(col).cloned().unwrap_or_else(unknown_error_value))
+            .collect()
+    } else {
+        // should never reach here due to `get_columns` above which will return
+        // empty columns if any value in the list is not a record
+        vec![Value::default(); headers.len()]
```
[the remark this answers is not in the document]
Comment (IanManske) [reply crates/nu-explore/src/nu_common/value.rs:161]: Yeah, it's a little awkward, but I didn't want to change the code too much. I just used the initial state for `rows` from the old code.

File: crates/nu-table/src/types/expanded.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sholderbach) [reply crates/nu-table/src/types/expanded.rs:123]: Ah I was mentally matching this against the `.unwrap_or_else` case.

**Pull request merged**